From 4859edfca7fb2408b8b9f90bf03c62e484dadc79 Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@nxp.com>
Date: Fri, 2 Feb 2018 11:02:27 +0200
Subject: [PATCH] MLK-17485: ASoC: fsl_sai: Specify supported rate_min and
 rate_max

Because fsl_sai_dai rates doesn't have a specific set of
rate values (.rates = SNDRV_PCM_RATE_KNOT) we need to provide
rate_min and rate_max otherwise functions trying to get
supported parameters will get confused and return an error.

Fixes:  1b6f0496e013 ("MLK-17428-8: ASoC: fsl_sai: support 768KHz sample rate")
Reviewed-by: Viorel Suman <viorel.suman@nxp.com>
Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
---
 sound/soc/fsl/fsl_sai.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/sound/soc/fsl/fsl_sai.c
+++ b/sound/soc/fsl/fsl_sai.c
@@ -917,6 +917,8 @@ static struct snd_soc_dai_driver fsl_sai
 		.stream_name = "CPU-Playback",
 		.channels_min = 1,
 		.channels_max = 32,
+		.rate_min = 8000,
+		.rate_max = 2822400,
 		.rates = SNDRV_PCM_RATE_KNOT,
 		.formats = FSL_SAI_FORMATS,
 	},
@@ -924,6 +926,8 @@ static struct snd_soc_dai_driver fsl_sai
 		.stream_name = "CPU-Capture",
 		.channels_min = 1,
 		.channels_max = 32,
+		.rate_min = 8000,
+		.rate_max = 2822400,
 		.rates = SNDRV_PCM_RATE_KNOT,
 		.formats = FSL_SAI_FORMATS,
 	},
